### PR TITLE
Simpler and less noisy make-clean

### PR DESCRIPTION
**Pull request by thockin**
Peeling commits out of other WIP efforts to get them out of the way.

/kind cleanup

```release-note
NONE
```

### CODE REVIEW

File: hack/make-rules/clean.sh [hunk 1]
```diff
@@ -24,15 +24,14 @@ source "${KUBE_ROOT}/hack/lib/util.sh"
 CLEAN_PATTERNS=(
   "_tmp"
   "doc_tmp"
-  "((?!staging\/src\/k8s\.io\/apiextensions-apiserver\/pkg\/generated\/openapi).)*/zz_generated.openapi.go"
```
Comment (BenTheElder) [inline hack/make-rules/clean.sh:27]: what happened with this pattern?

**Review by BenTheElder — COMMENTED** [review]
/lgtm
/hold
Generally LGTM, one question ...

File: hack/make-rules/clean.sh (hunk 1 above)
Comment (thockin) [reply hack/make-rules/clean.sh:27]: It matches nothing - all openapi files are properly checked in now (for a while, actually)

**Pull request merged**